invoices/invoiceregistry.go

```diff
@@ -1676,6 +1661,9 @@ func (i *InvoiceRegistry) notifyHodlSubscribers(htlcResolution HtlcResolution) {
 func (i *InvoiceRegistry) hodlSubscribe(subscriber chan<- interface{},
 	circuitKey channeldb.CircuitKey) {
 
+	i.Lock()
```

Comment:
nit: could you update commit message explaining what we were previously covering with this lock / why it's safe to do? Difficult to follow in review

Reply:
sorry about the confusion. The original idea was to remove unnecessary areas being locked. For instance, when sending over a go channel or performing a db tx. This commit has now been removed to focus on one task only.